libcxx/src/shared_recursive_mutex.cpp

```diff
@@ -0,0 +1,122 @@
+#include <cassert>
+#include <shared_recursive_mutex>
+
+constexpr uint32_t SHARED_USER_MAX_CNT = UINT32_MAX;
+constexpr pthread_t THREAD_ID_NOEXIST  = 0;
```

Comment:
If it's intended to support non-pthread platforms, `__thread_id` should be used. But it's a bit unfortunate that it's default constructor is not constexpr, and I _think_ we can drop this constant and use `__thread_id{}` instead.